packages/core/src/state-management/state-manager.ts

```diff
@@ -323,7 +324,7 @@ export class StateManager {
   /**
    * Request anchor for the latest stream state
    */
-  anchor(state$: RunningState): Subscription {
+  async anchor(state$: RunningState): Promise<Subscription> {
```

Comment:
The request to store the anchor should be part of the rxjs pipeline, IMO, retaining sync method signature.